openforcefield/typing/engines/smirnoff/forcefield.py

```diff
@@ -247,31 +277,58 @@ def _initialize(self):
         )  # ParameterIO classes to be used for each file type
         self._parameters = ParameterList(
         )  # ParameterHandler objects instantiated for each parameter type
+        self._aromaticity_model = None
+        #self._used_aromaticity_models = set(
+        #)  # Keep track of the aromaticity model(s) this ForceField uses
+
 
-    # TODO : This is unused and shouldn't be here -- Move to parameteriohandler
-    def _parse_version(self, root):
+    def _check_smirnoff_version_compatibility(self, version):
         """
-        Parse the forcefield version number and make sure it is supported.
+        Raise a parsing exception if the given file version is incompatible with this ForceField class.
 
         Parameters
         ----------
-        root : etree.Element
-            The document root
+        version : str
+            The SMIRNOFF version being read.
+
+        Raises
+        ------
+        SMIRNOFFVersionError if an incompatible version is passed in.
 
         """
-        if 'version' in root.attrib:
-            version = root.attrib['version']
-            # Use PEP-440 compliant version number comparison, if requested
-            if (not self.disable_version_check) and (
-                    packaging.version.parse(version) >
-                    packaging.version.parse(MAX_SUPPORTED_VERION)):
-                self._raise_parsing_exception(
-                    root,
-                    'SMIRNOFF offxml file was written with version %s, but this version of ForceField only supports up to version %s'
-                    % (self.version, MAX_SUPPORTED_VERSION))
-        else:
-            self._raise_parsing_exception(
-                root, "'version' attribute must be specified in SMIRNOFF tag")
+        import packaging.version
+        # Use PEP-440 compliant version number comparison, if requested
+        if (not self.disable_version_check) and (
+                packaging.version.parse(str(version)) >
+                packaging.version.parse(str(self._MAX_SUPPORTED_SMIRNOFF_VERSION))):
+            raise SMIRNOFFVersionError(
+                'SMIRNOFF offxml file was written with version {}, but this version of ForceField only supports '
+                'up to version {}'.format(version, self._MAX_SUPPORTED_SMIRNOFF_VERSION))
+
+
+    def _register_aromaticity_model(self, aromaticity_model):
```

Comment:
Thanks for the feedback, however I don't think we want to treat aromaticity model as a property. 

The difference is that multiple OFFXMLs (or SMIRNOFF data sources) can be read into a single forcefield. The first one read should set the ForceField's aromaticity model, and each subsequent one will have its aromaticity model checked against the first's. I prefer directing all aromaticity models through this one function, but because multiple aromaticity models can be read, I think it would be confusing to handle it with a property setter.

Reply:
What I mean is that `register` implies that multiple aromaticity models can be registered. If this is true, +1 (although I'd use `add` in this case instead of `register`). If a `ForceField` instead can have only 1 aromaticity model, then `set` is probably the standard word. This is an internal-usage method anyway so not that important.

Reply:
Ah, I agree. Changed it to `ForceField._set_aromaticity_model` in 7a61084